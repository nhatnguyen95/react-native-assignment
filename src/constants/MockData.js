export const trips = [
  {
    trip_id: 17094,
    duration: 1956,
    distance: 16,
    start_time: "2017-07-04T07:50:36+0000",
    end_time: "2017-07-04T08:23:12+0000",
  },
  {
    trip_id: 17095,
    duration: 891,
    distance: 4,
    start_time: "2017-07-04T08:30:45+0000",
    end_time: "2017-07-04T08:45:36+0000",
  },
];
