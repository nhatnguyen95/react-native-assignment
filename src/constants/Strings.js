const DATE_TIME_FORMAT = "DD/MM/YYYY HH:mm:ss";
export default {
  DATE_TIME_FORMAT,
  DURATION: 'Duration',
  DISTANCE: 'Distance',
  START_TIME: 'Start Time',
  END_TIME: 'End Time',
  SECONDS: 'second(s)',
  KILOMETERS: 'kilometer(s)',
  TRIP_ID: 'TRIP',
  TWO_TRIPS: 'TWO TRIPS',
  EMPTY_TRIP: 'EMPTY TRIP',
  BIG_TRIPS: 'BIG TRIPS',
  DATA_NOT_FOUND : 'Data not found!',
};
