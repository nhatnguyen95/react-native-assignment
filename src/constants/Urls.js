BASE_URL = "https://api.gigacover.com/";

RECRUITMENT_CHALLENGE = "recruitment/challenge";

export default {
  BASE_URL,
  RECRUITMENT_CHALLENGE,
}