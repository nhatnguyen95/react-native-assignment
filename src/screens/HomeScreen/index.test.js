import "react-native";
import React from "react";
import { HomeScreen } from "./index";

import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { trips } from "constants/MockData";


describe("HomeScreen", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const wrapper = renderer.create(<HomeScreen data={trips}/>).toJSON();
      expect(wrapper).toMatchSnapshot();
    });

    it("should render all items in trips", () => {
      const wrapper = shallow(<HomeScreen data={trips} />);
      const flatList = wrapper.find('FlatList').props();
      expect(flatList.data.length).toBe(2);
    });
  });

  it("getData function should be called during componentDidMount", () => {
    const getData = jest.fn();
    renderer.create(<HomeScreen getData={getData} />);
    expect(getData).toHaveBeenCalled();
  });

  describe("Navigate to other Screen", () => {
    it("should navigate to other screen", () => {
      const tripIndex = 0;
      const navigation = { navigate: jest.fn() };
      const wrapper = renderer.create(<HomeScreen navigation={navigation} />);
      wrapper.getInstance().onPressItem(tripIndex);
      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
