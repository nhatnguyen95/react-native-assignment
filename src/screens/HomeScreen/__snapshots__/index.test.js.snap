// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "distance": 16,
          "duration": 1956,
          "end_time": "2017-07-04T08:23:12+0000",
          "start_time": "2017-07-04T07:50:36+0000",
          "trip_id": 17094,
        },
        Object {
          "distance": 4,
          "duration": 891,
          "end_time": "2017-07-04T08:45:36+0000",
          "start_time": "2017-07-04T08:30:45+0000",
          "trip_id": 17095,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "justifyContent": "center",
              "marginTop": 1,
              "opacity": 1,
              "paddingVertical": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Trip Id: 
            17094
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "justifyContent": "center",
              "marginTop": 1,
              "opacity": 1,
              "paddingVertical": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Trip Id: 
            17095
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
